### DISCUSSION TITLE
preact/compat does the take up the bundle size in webpack?

### DISCUSSION THREAD

**Discussion by MTtankkeo** [preactjs/preact#4468] [Q&A] [answered]
preact/compat is a sub module...?, so if i refer to it, does it occupy a bundle size by included to bundle?

> I was suddenly curious because I'm using preact a lot, haha 🤣

**Comment by rschristian** [comment, marked as answer]
Yep, if you use `preact/compat` in any way, it'll be added to the overall bundle of your app.

Is there a particular problem you're having?

**Comment by MTtankkeo** [comment]
Thanks, I was just curious, because I like to lightweight my project more.

**Reply by rschristian** [reply]
Along those lines I can supply a bit more info:

Firstly, `preact/compat` is side-effectful, meaning using any bit of it (`import { memo } from 'preact/compat'`) drags in all of `preact/compat` -- it cannot be treeshaken due to how it needs to work. So if you do use it, you can't reduce the size of it at all unfortunately.

However, simply adding the aliases to your project doesn't mean `preact/compat` is used. The aliases will only result in `preact/compat` being added if you're actually using something that tries to import from `react` or `react-dom` -- if nothing does, the aliases will be unused, and `preact/compat` shouldn't end up in your app (assuming you're not directly using it).

**Reply by MTtankkeo** [reply]
Thanks, I'm going to divide my react packages into separate react and preact because of that problem.

**Reply by rschristian** [reply]
That'd be ideal.

In case it's helpful, I wrote [a tiny demo](https://github.com/rschristian/react-preact-lib) a while back showing how to do it. Not that you have to follow it, but it's an implementation of this idea.

**Reply by MTtankkeo** [reply]
I was going to write the source code separately for both of packages, but that's very very great! 😋 Thanks.

> My fingers would hurt without your demo 🤣
